program-runtime/src/loaded_programs.rs

```diff
@@ -382,6 +382,22 @@ impl LoadedProgram {
     }
 
     pub fn to_unloaded(&self) -> Option<Self> {
+        match &self.program {
+            LoadedProgramType::LegacyV0(_)
+            | LoadedProgramType::LegacyV1(_)
+            | LoadedProgramType::Typed(_) => {}
+            #[cfg(test)]
+            LoadedProgramType::TestLoaded(_) => {}
+            LoadedProgramType::FailedVerification(_)
+            | LoadedProgramType::Closed
+            | LoadedProgramType::DelayVisibility
+            | LoadedProgramType::Unloaded(_)
+            | LoadedProgramType::Builtin(_) => {
+                debug_assert!(false);
```

Comment:
We don't need nor the assert nor the error, see unload_program_entry(). It'll
call to_unloaded for all the entries in there, and unload only the ones for
which we return Some. Returning None is fine by design.